### ISSUE TITLE
unexpected error TS2322: Type, Is this a bug?

### ISSUE THREAD

**Issue by mokeyish** [Microsoft/TypeScript#17035]
Is this a bug ?
The function I write：
![baidushurufa_2017-7-9_3-13-15](https://user-images.githubusercontent.com/16131917/27988447-958f85cc-6454-11e7-9332-e905e75cea22.png)
Here is array.some signature.
![baidushurufa_2017-7-9_3-14-36](https://user-images.githubusercontent.com/16131917/27988462-ccbe6284-6454-11e7-9ad3-3b880b6e1914.png)
The error in IDE output window.
error TS2322: Type '(predicate: any) => boolean' is not assignable to type '{ (): boolean; (predicate: (value?: any, index?: number, Array?: any[]) => boolean): boolean; }'.

I change code  like this :
![baidushurufa_2017-7-9_3-23-13](https://user-images.githubusercontent.com/16131917/27988521-15a99e18-6456-11e7-809d-dc216d1c1bb8.png)

The error will be also changed.
error TS2322: Type '(predicate: (this: void, value: any, index: number, array: any[]) => boolean) => boolean' is not assignable to type '{ (): boolean; (predicate: (value?: any, index?: number, Array?: any[]) => boolean): boolean; }

**Comment by bcherny** [comment]
What is your typing for `Array.prototype.Any`? This question probably belongs on StackOverflow.

**Comment by mokeyish** [comment]
@bcherny this is a series of linq method ,I have been add 'First' 'LastOrDefault' .etc into Array prototype  and it works well.but in 'Any' ,it shows error TS2322.it seems to take 'this.some(predicate)' as a whole returnd value .

**Comment by ikatyang** [comment]
@mokeyish 

Before you create an new issue, it'd be better to search first, you'll probably find something related, like [this](https://github.com/Microsoft/TypeScript/issues/16192).

And yes, this is an issue-tracker, not a support forum, it'd better to ask something like this at StackOverflow next time unless you can confirm this is a bug, and make sure you fill in the [template](https://raw.githubusercontent.com/Microsoft/TypeScript/master/issue_template.md) so that we could easily find out what your problem is.

```ts
declare global {
    interface Array<T> {
        any(callbackfn?: (this: void, value: T, index: number, array: T[]) => boolean): boolean;
        any(callbackfn: (this: void, value: T, index: number, array: T[]) => boolean, thisArg: undefined): boolean;
        any<Z>(callbackfn: (this: Z, value: T, index: number, array: T[]) => boolean, thisArg: Z): boolean;
    }
}

Array.prototype.any = function <T>(this: T[], callbackfn?: (value: T, index: number, array: T[]) => boolean, thisArg?: any) {
    return callbackfn ? this.some(callbackfn, thisArg) : this.length > 0;
};

export {};
```

**Comment by RyanCavanaugh** [comment]
Just going to assume this is a question. As a starting point, we are not blessed with the time needed to manually type in code from screenshots to try to answer your question. Please post text and be clear about why you think it is a bug if you're going to be posting on the bug tracker.
